Apps/DiceRoller/DiceRoller/ISDRProjectManager.m

```diff
@@ -24,15 +24,77 @@ - (id)initWithNibName:(NSString *)nibNameOrNil bundle:(NSBundle *)nibBundleOrNil
     return self;
 }
 
+
 - (void)viewWillAppear:(BOOL)animated {
     
     [super viewWillAppear:animated];
     
     dm = [DataManager getInstance];
-    int curProjID = [dm getProjectID];
-    NSString *curProjIDStr = (curProjID > 0) ? [NSString stringWithFormat:@"%d", curProjID] : kNO_PROJECT;
     
-    [projectLbl setText:[NSString stringWithFormat:@"Uploading to Project: %@", curProjIDStr]];
+    pm = [ [ISDRProjectManager alloc] init];
+    
+    bool projectValid = [pm projectHasValidFields];
+    
+    if( projectValid ){
+        
+        int curProjID = [dm getProjectID];
+        NSString *curProjIDStr = (curProjID > 100) ? [NSString stringWithFormat:@"%d", curProjID] : kNO_PROJECT;
+        [projectLbl setText:[NSString stringWithFormat:@"Uploading to Project: %@", curProjIDStr]];
+        
+    }else{
```

Comment:
Me being picky:
Line 44 ISDRProjectManager.m should be:
```} else {```
- Spaces between else and the curly brackets